scripts/bumpfee.py

```diff
@@ -0,0 +1,279 @@
+            if wallet.is_known_script(vout.scriptPubKey) and fee > 0:
+                # check if the output is a change address
+                if wallet.script_to_path(vout.scriptPubKey)[-2] != 1:
+                    continue
+
+                # deduct fee from the change
+                tx.vout[ix].nValue -= fee
+                fee = 0
+
+                # if the output value is less than zero, remove it
+                if tx.vout[ix].nValue < 0:
+                    jlog.info("Dynamically calculated change lower than zero; dropping.")
+
+                    # update fee to the additional amount needed to pay for the tx
+                    # accounting for the removal of the output
+                    fee = abs(tx.vout[ix].nValue) - len(tx.vout[ix].serialize())
+                    tx.vout.remove(vout)
+                    continue
+
+                # if the output value is below the dust threshold, remove it
+                if tx.vout[ix].nValue <= jm_single().BITCOIN_DUST_THRESHOLD:
+                    jlog.info("Dynamically calculated change lower than dust: " +
+                        btc.amount_to_str(tx.vout[ix].nValue) + "; dropping.")
+                    tx.vout.remove(vout)
+                break
+
+    # create new transaction from the old
+    # there's the possibility that it returns the same transaction as the old
+    # if no outputs were available to deduct a fee from
+    return btc.CMutableTransaction(
+        tx.vin, tx.vout, nLockTime=tx.nLockTime,
+        nVersion=tx.nVersion)
+
+if __name__ == '__main__':
+    (options, args) = parser.parse_args()
+    load_program_config(config_path=options.datadir)
+    if len(args) < 2:
+        parser.error("JoinMarket bumpfee needs arguments:"
+                     " wallet file and txid.")
+        sys.exit(EXIT_ARGERROR)
+
+    wallet_name = args[0]
+    txid = args[1]
+
+    # If tx_fees are set manually by CLI argument, override joinmarket.cfg:
+    if int(options.txfee) > 0:
+        jm_single().config.set("POLICY", "tx_fees", str(options.txfee))
+    fee_per_kb = jm_single().bc_interface.estimate_fee_per_kb(
+        jm_single().config.getint("POLICY", "tx_fees"))
+    if fee_per_kb is None:
+        raise RuntimeError("Cannot estimate fee per kB, possibly" +
+                           " a failure of connection to the blockchain.")
+
+    # open the wallet and synchronize it
+    wallet_path = get_wallet_path(wallet_name, None)
+    wallet = open_test_wallet_maybe(
+        wallet_path, wallet_name, options.amtmixdepths - 1,
+        wallet_password_stdin=options.wallet_password_stdin,
+        gap_limit=options.gaplimit)
+    wallet_service = WalletService(wallet)
+    if wallet_service.rpc_error:
+        sys.exit(EXIT_FAILURE)
+    while not wallet_service.synced:
+        wallet_service.sync_wallet(fast=not options.recoversync)
+    wallet_service.startService()
+
+    orig_tx = wallet_service.get_transaction(hextobin(txid))
```

Comment:
Put a check for a `None` return, because that can happen if the txid isn't recognized as belonging to the wallet.